src/include/btmem.h

```diff
@@ -28,11 +28,12 @@
 /* AUTOMATIC FLAG VALUE GENERATION START */
 #define WT_REC_CHECKPOINT 0x01u
 #define WT_REC_EVICT 0x02u
-#define WT_REC_HS 0x04u
-#define WT_REC_IN_MEMORY 0x08u
-#define WT_REC_SCRUB 0x10u
-#define WT_REC_VISIBILITY_ERR 0x20u
-#define WT_REC_VISIBLE_ALL 0x40u
+#define WT_REC_EXPECT_CLEAN 0x04u
```

Comment:
It took me a while to figure out what this flag meant. I was initially confused because I thought it meant the page was clean coming in, and that doesn't make sense for reconciliation.

Does it need a flag at all? Could we instead add a new WT_BTREE::syncing state of WT_BTREE_SYNC_CLOSE (or possibly reuse the existing WT_BTREE_SYNC_RUNNING - it seems as though blocking out other threads from allocating space in the file while evicting a file would be valid.

Reply:
As discussed, I'll have a try.

Reply:
We can't reusing WT_BTREE_SYNC_RUNNING as we also set this flag in all checkpoints not only when we are closing. It is a bit risky to introduce a new SYNC flag. Therefore, we prefer to introduce a new reconciliation flag in this case. As requested, I renamed the flag to WT_REC_NO_CACHE_RESTORE. Hopefully that is clearer.